src/lib.rs

```diff
@@ -23,20 +23,51 @@ impl From<key::EitherEd25519SecretKey> for PrivateKey {
     }
 }
 
+use rand::rngs::OsRng;
+use rand_chacha::ChaChaRng;
+use rand_core::SeedableRng;
+
 #[wasm_bindgen]
 impl PrivateKey {
     pub fn from_bech32(bech32_str: &str) -> Result<PrivateKey, JsValue> {
         crypto::SecretKey::try_from_bech32_str(&bech32_str)
             .map(key::EitherEd25519SecretKey::Extended)
-            .or_else(|_| crypto::SecretKey::try_from_bech32_str(&bech32_str)
-                .map(key::EitherEd25519SecretKey::Normal))
+            .or_else(|_| {
+                crypto::SecretKey::try_from_bech32_str(&bech32_str)
+                    .map(key::EitherEd25519SecretKey::Normal)
+            })
             .map(PrivateKey)
             .map_err(|_| JsValue::from_str("Invalid secret key"))
     }
 
     pub fn to_public(&self) -> PublicKey {
         self.0.to_public().into()
     }
+
+    pub fn generate_ed25519() -> Result<PrivateKey, JsValue> {
+        OsRng::new()
+            .and_then(ChaChaRng::from_rng)
```

Comment:
probably overkill to use ChaChaRng, OsRng should suffice in this case, specially as this is used only once.